packages/data-service/src/data-service.ts

```diff
@@ -153,6 +153,28 @@ export type UpdatePreview = {
   changes: UpdatePreviewChange[];
 };
 
+export type StreamProcessor = {
+  id: string;
+  name: string;
+  state:
+    | 'CREATING'
+    | 'CREATED'
+    | 'VALIDATING'
+    | 'PROVISIONING'
+    | 'RECEIVED_ON_DISPATCHER'
+    | 'STARTING'
+    | 'STARTED'
+    | 'STOPPING'
+    | 'STOPPED'
+    | 'RELEASING'
+    | 'DROPPING'
+    | 'DROPPED'
+    | 'FAILED';
+  pipeline: Document[];
+  lastStateChange: Date;
+  lastModified: Date;
```

Comment:
Are we sure that we will be receiving date object from the command run? If not, maybe we should type these as string or number (in case of timestamp)?

Reply:
Good call, they are shown as below in playground result but doesn't seem to be an instance of `Date`. Will change to `string`.
```
    "lastModified": {
      "$date": "2023-12-04T04:06:49.506Z"
    },
```

Reply:
Hmmm, I think the format you're seeing here is not a `string` but a EJSON representation of a bson Date value. I think you should be able to pass `promoteValues` to the command options and then the driver should actually promote this value to a JavaScript Date:

![image](https://github.com/mongodb-js/compass/assets/5036933/0fe6e423-5722-4a53-a3f0-4cf4132d2812)

This is something we also do for some other operations, so maybe would make sense here too so that the consumers of this method don't need to deal with EJSON. Don't have a preference though if we want to do this. If not, this should probably be a Document type, not a string though.

Reply:
It looks like `promoteValues` is `true` by default and I can confirm that the value returned from driver is actually an instance of `Date`. So I've changed the type back to `Date`.

Reply:
Awesome, thanks for double-checking